### DISCUSSION TITLE
Allowing more characters in set names

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/webwork2#2332] [General]
An exercise set can only have the following characters in its set_id (which is a tinyblob): letters, digits, dashes, underscores, periods, and spaces.

Can this be extended? In practice, I have found I wanted to use a comma, a colon, and parentheses at various times. I could imagine wanting certain other things, not to mention accented characters in a language that used them.

I get it with a comma, that it would mess up CSV files, but we could change the CSV grade files to wrap the set names in double quotes. Where else do we have issues like this?

**Comment by drgrice1** [comment]
Commas go much further than CSV files.  That would also mess up the entire scheme for gateway sets with versions.

For other characters there are also complications with file names (the set id is used for the directory when a new problems is added), and with the set id being used in urls.

**Reply by Alex-Jordan** [reply]
I see. Bummer. If only we were designing a new database structure...

**Reply by drgrice1** [reply]
I think that it might be possible to do something about colons and parentheses. Although we would need to tread carefully here.  I think there are some other database considerations with those characters as well.
